docs/help/contributor/style-guide/documenting-code.md

```diff
@@ -0,0 +1,212 @@
+    Create the service by running:
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+=== ":no_entry: Incorrect"
+    **Example 1:**
+
+    Create the service:
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+    **Example 2:**
+
+    Run:
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+    **Example 3:**
+
+    Run the command below to create a service:
+
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+
+## Documenting YAML
+>Use **kubectl apply** for files/objects that the user creates — it works for both “create” and “update”, and the source of truth is their local files.
+Use **kubectl edit** for files which are shipped as part of the Knative software, like the serving/eventing ConfigMaps.
+
+=== ":white_check_mark: Correct"
+
+    Creating a new file:
+    ```
+    kubectl apply -f - <<EOF
+    # code
+    EOF
+    ```
+
+    Editing a file:
+    ```
+    kubectl -n <namespace> edit cm <filename>
+    ```
+
+=== ":no_entry: Incorrect"
+    ```
+    cat <<EOF | kubectl create -f -
+    # code
+    EOF
+    ```
+
+
+## Words requiring code formatting
+Use code formatting to indicate special purpose text. Apply code formatting to the following content:
+
+* Text that the user enters into a field
+* Filenames and path names
+* Configuration file content
+* Sections of code
+* Programming examples
+* Command line text
+* Environment variable text
+
+
+## Referencing Variables in Code Blocks
+
+>Format variables in code blocks like so: <service-name>
+
+> - All lower case
+- Hyphens between words
+- Explanation for each variable below code block
+- Explanation format is “Where... `<service-name>` is…"
+- If there are multiple variables, see below.
+
+### Single Variable
+=== ":white_check_mark: Correct"
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+=== ":no_entry: Incorrect"
+    ```bash
+    kn create service {SERVICE_NAME}
+    ```
+    {SERVICE_NAME} = The name of your service
+
+
+### Multiple Variables
+
+=== ":white_check_mark: Correct"
+    ```bash
+    kn create service <service-name> --revision-name <revision-name>
+    ```
+    Where:
+
+    * `<service-name>` is the name of your Knative Service.
+    * `<revision-name>` is the desired name of your revision
+
+=== ":no_entry: Incorrect"
+    ```bash
+    kn create service <service-name> --revision-name <revision-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.<br>
+    Where `<revision-name>` is the desired name of your revision
+
+
+## CLI Input and Output
+### CLI Input
+> Be sure to specify the language your code is in, bash for any CLI commands.
```

Comment:
`bash` or `shell` for CLI? I see both throughout docs, what's the difference?

Reply:
I have standardized on bash... I think either works about the same? Bash is an implementation of shell, but in practice they are used interchangeably

Reply:
They are the same
https://pygments.org/docs/lexers/#lexers-for-various-shells
```
Short names
bash, sh, ksh, zsh, shell
```